### PR TITLE
RavenDB-7067 Use HttpCompletionOption.ResponseHeadersRead in HttpClient in order to avoid OutOfMemoryException with big response

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/AttachmentHandler.cs [hunk 1]
```diff
@@ -102,6 +102,7 @@ public Task GetPost()
                         HttpContext.Response.Headers.Remove("Content-Type");
                 }
                 HttpContext.Response.Headers["Content-Hash"] = attachment.Base64Hash.ToString();
+                HttpContext.Response.Headers["Content-Size"] = attachment.Stream.Length.ToString();
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/AttachmentHandler.cs:105]: This is problematic. It looks like we meant `Content-Length` but put something else there.
Better to put `Total -Attachment-Size` doesn't look like a known header

File: src/Raven.Client/Documents/Commands/HeadDocumentCommand.cs [hunk 2]
```diff
@@ -18,6 +17,7 @@ public HeadDocumentCommand(string id, long? etag)
         {
             _id = id ?? throw new ArgumentNullException(nameof(id));
             _etag = etag;
+            ResponseType = RavenCommandResponseType.Raw;
```
Comment (ayende) [inline src/Raven.Client/Documents/Commands/HeadDocumentCommand.cs:20]: Why `Raw` and not `Empty`? To prevent caching?

Comment (fitzchak) [reply src/Raven.Client/Documents/Commands/HeadDocumentCommand.cs:20]: We should support cache, fixed: https://github.com/ravendb/ravendb/pull/2368

File: src/Raven.Server/Documents/Handlers/AttachmentHandler.cs (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Server/Documents/Handlers/AttachmentHandler.cs:105]: How about `Attachment-Size`?

File: src/Raven.Client/Documents/Commands/DeleteDocumentCommand.cs [hunk 3]
```diff
@@ -17,6 +17,7 @@ public DeleteDocumentCommand(string id, long? etag)
 
             _id = id;
             _etag = etag;
+            ResponseType = RavenCommandResponseType.Empty;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Commands/DeleteDocumentCommand.cs:20]: Maybe RavenCommand can set ResponseType to Empty automatically? And RavenCommand<T> to Object?

Comment (fitzchak) [reply src/Raven.Client/Documents/Commands/DeleteDocumentCommand.cs:20]: I will make this change.

File: src/Raven.Server/Documents/Handlers/AttachmentHandler.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Handlers/AttachmentHandler.cs:105]: Maybe change both? Content-Hash -> Attachment-Hash and Content-Size to Attachment-Size?

File: src/Sparrow/Collections/LockFree/ConcurrentDictionary.cs [hunk 4]
```diff
@@ -641,7 +641,7 @@ IEnumerator IEnumerable.GetEnumerator()
 
         IDictionaryEnumerator IDictionary.GetEnumerator()
         {
-            return _table.GetdIDictEnumerator();
+            return _table.GetDictionayEnumerator();
```
Comment (ppekrol) [inline src/Sparrow/Collections/LockFree/ConcurrentDictionary.cs:644]: We have a typo here.

File: src/Raven.Server/Documents/Handlers/AttachmentHandler.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Documents/Handlers/AttachmentHandler.cs:105]: I Like that, yes

File: src/Raven.Client/Documents/Commands/DeleteDocumentCommand.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Commands/DeleteDocumentCommand.cs:20]: Thanks.

I've meant RavenCommand -> Empty and RavenCommand[T] -> Object. I think that the parenthesis were stripped.

File: src/Sparrow/Collections/LockFree/ConcurrentDictionary.cs (hunk 4 above)
Comment (fitzchak) [reply src/Sparrow/Collections/LockFree/ConcurrentDictionary.cs:644]: Fixed

File: src/Raven.Client/Documents/Commands/DeleteDocumentCommand.cs (hunk 3 above)
Comment (fitzchak) [reply src/Raven.Client/Documents/Commands/DeleteDocumentCommand.cs:20]: Done: https://github.com/ravendb/ravendb/pull/2371

File: src/Raven.Server/Documents/Handlers/AttachmentHandler.cs (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Server/Documents/Handlers/AttachmentHandler.cs:105]: Done: https://github.com/ravendb/ravendb/pull/2371

**Pull request merged**